library/hawkular_alerts_group_dampening.py

```diff
@@ -0,0 +1,297 @@
+            self.module.fail_json(msg="Failed to get group trigger dampenings. Error: {error}".format(error=e))
+        return next((True for d in current_dampenings if d.dampening_id == dampening_id), False)
+
+    def delete_group_dampenings(self, group_id, dampenings):
+        """ Deletes group dampenings in Hawkular Alerting component
+
+            Returns:
+                whether or not a change took place and a short message
+                describing the operation executed
+        """
+        if not self.group_trigger_exist(group_id):
+            self.module.fail_json(msg="Group trigger {group_id} doesn't exist".format(group_id=group_id))
+        messages = []
+        for dampening in dampenings:
+            try:
+                self.client.delete_group_dampening(group_id, dampening["dampening_id"])
+                self.changed = True
+                messages.append("Successfully deleted group dampening {dampening_id}".format(dampening_id=dampening["dampening_id"]))
+            except urllib2.HTTPError as e:
+                if e.code == 404:
+                    messages.append("Group dampening {dampening_id} doesn't exist".format(dampening_id=dampening["dampening_id"]))
+                else:
+                    self.module.fail_json(msg="Failed to delete group dampening. Error: {error}".format(error=e))
+            except Exception as e:
+                self.module.fail_json(msg="Failed to delete group dampening. Error: {error}".format(error=e))
+        return dict(
+            msg=messages,
+            changed=self.changed)
+
+    def list_group_dampenings(self, group_id):
+        """ Lists all dmapenings in the group trigger
+
+        Returns:
+            whether or not a change took place and a short message
+            describing the operation executed
+        """
+        if not self.group_trigger_exist(group_id):
+            self.module.fail_json(msg="Group trigger {group_id} doesn't exist".format(group_id=group_id))
+        group_dampenings = []
+        try:
+            group_dampenings = self.client.list_dampenings(group_id)
+            group_dampenings_dicts_list = [vars(dampening) for dampening in group_dampenings]
+        except Exception as e:
+            self.module.fail_json(msg="Failed to list group dampenings. Error: {error}".format(error=e))
+        return dict(
+            msg="Successfully listed group dampenings",
+            changed=self.changed,
+            group_dampenings=group_dampenings_dicts_list)
+
+    def update_required(self, group_id, desired_dampening):
+        """ Checks whether an update is required for the group trigger
+
+            Returns:
+               True, if an update is required for the dampening, False otherwise
+        """
+        current_dampenings = self.client.list_dampenings(group_id)
+        for current_dampening in current_dampenings:
+            if current_dampening.trigger_mode == desired_dampening["trigger_mode"]:
```

Comment:
> Is the idea that there is at most one dampening with given mode?

yes, one dampening per trigger mode (max of 2 - `FIRING` and `AUTORESOLVE`)

> If so, what you think of changing dampenings module param to a dict keyed by trigger mode?

Interesting proposal! it will make my life easier, I'm wondering if it will complicate the user, what do you think?

> EDIT:  elsewhere you match by dampening_id. Shouldn't it consistently be one (id or mode) everywhere?

Sorry, I don't understand your question